code/modules/ai/ai_behaviors/human_mobs/helper_procs.dm

```diff
@@ -0,0 +1,128 @@
+//Interaction stuff
+
+///Checks if the AI can or should use this
+/obj/item/proc/ai_should_use(mob/living/target, mob/living/user)
+	return TRUE
+
+/obj/item/stack/medical/heal_pack/ai_should_use(mob/living/target, mob/living/user)
+	if(!ishuman(target))
+		return FALSE
+	var/mob/living/carbon/human/human_mob = target
+	for(var/limb AS in human_mob.limbs)
+		if(can_heal_limb(limb))
+			return TRUE
+	return FALSE
+
+/obj/item/reagent_containers/ai_should_use(mob/living/target, mob/living/user)
+	for(var/datum/reagent/reagent AS in reagents.reagent_list)
+		if(reagent.volume + target.reagents.get_reagent_amount(reagent.type) > reagent.overdose_threshold)
+			return FALSE
+	return TRUE
+
+/obj/item/reagent_containers/hypospray/ai_should_use(mob/living/target, mob/living/user)
+	if(!length(reagents.reagent_list)) //todo: discard if empty
+		return FALSE
+	for(var/datum/reagent/reagent AS in reagents.reagent_list)
+		if((reagent.volume / reagents.total_volume * amount_per_transfer_from_this) + target.reagents.get_reagent_amount(reagent.type) > reagent.overdose_threshold)
+			return FALSE
+	return TRUE
+
+///AI uses this item in some manner, such as consuming or activating it
+/obj/item/proc/ai_use(mob/living/target, mob/living/user)
+	return FALSE
+
+/obj/item/stack/medical/ai_use(mob/living/target, mob/living/user)
+	attack(target, user)
+
+/obj/item/reagent_containers/ai_use(mob/living/target, mob/living/user)
+	afterattack(target, user, TRUE) //why are all these medical item inconsistant with their proc usage, holy shit
+
+/obj/item/reagent_containers/pill/ai_use(mob/living/target, mob/living/user)
+	attack(target, user)
+
+/obj/item/tweezers/ai_use(mob/living/target, mob/living/user)
+	attack(target, user)
+
+/obj/item/tweezers_advanced/ai_use(mob/living/target, mob/living/user)
+	attack(target, user)
+
+///AI mob interaction with this atom, such as picking it up
+/atom/proc/do_ai_interact(mob/living/interactor)
+	interactor.UnarmedAttack(src, TRUE) //only used for picking up items atm
+
+//weapon engagement range
+
+///Optimal range for AI to fight at, using this weapon
+/obj/item/weapon/proc/get_ai_combat_range()
+	return list(0, 1)
+
+/obj/item/weapon/twohanded/spear/get_ai_combat_range()
+	return 2
```

Comment:
Unrelated But I thought we said we should kill just unlabeled list and int mixing like this what happen

Reply:
I did? It now uses a min and max range.
SETTING the min and max range is just using min() and max() so whether its a 1 item list or not, it still works fine